### PR TITLE
Add Log Wrapper to federator

### PR DESCRIPTION
**Pull request by RafilxTenfen**
### Description
- Add log wrapper to federator to talk with the log4js
- Now the only place where it should import the log4js is in the `logs.ts` and `logWrapper.ts` files
- Add logs to federator to help with debugging errors
- Add context to logs
- Add nodemon to keep watching changes in files while developing 

### How to Test

- Run both `ganache-cli`
- Enter into the bridge directory and run all the tests

#### Case 1
1. Go to bridge directory
```shell
$~ cd bridge
```

2. Run the first ganache-cli
```shell
$~ npm run ganache
```

3. Open another shell and run the ganache mirror
```shell
$~ npm run ganache-mirror
```
4. Go to the federator directory
```shell
$~ cd federator
```

5. Run the federator node
```shell
$~ npm run start-dev
```

__Expected Result__
image

#### Case 2

1. Go to bridge directory
```shell
$~ cd bridge
```

2. Run all the tests
```shell
$~ npm run test
```

__Expected Result__
- It should pass the test
image

3. Run the first ganache-cli
```shell
$~ npm run ganache
```

4. Open another shell and run the ganache mirror
```shell
$~ npm run ganache-mirror
```

5. Go to the federator directory
```shell
$~ cd ../federator
```

6. Run the integration test
```shell
$~ npm run integrationTest
```

__Expected Result__
- It should pass the test
image

7. Run the NFT integration test
```shell
$~ npm run nftIntegrationTest
```
__Expected Result__
- It should pass the test
image

#### Bridge Directory `cd bridge`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Contracts are compiling `npm run compile`

#### Federator Directory `cd federator`
- [x] Lint is clean `npm run lint`
- [x] Test is passing `npm run test`
- [x] Typescript is compiling `npm run build`

### CODE REVIEW

File: federator/src/lib/logs.ts [hunk 1]
```diff
@@ -1 +1,86 @@
-export class Logs {}
+import log4js from 'log4js';
+import { LogWrapper } from './logWrapper';
+import logConfig from '../../config/log-config.json';
+
+export const LOGGER_CATEGORY_TEST_INTEGRATION = 'test';
+export const LOGGER_CATEGORY_TEST_FEDERATOR = 'FEDERATOR';
```
Comment (pmprete) [inline federator/src/lib/logs.ts:6]: maybe rename it to TEST_FEDERATOR?

File: federator/src/lib/logs.ts [hunk 2]
```diff
@@ -1 +1,86 @@
...
+export const LOGGER_CATEGORY_TEST_FEDERATOR_NFT = 'NFT FEDERATOR';
+export const LOGGER_CATEGORY_HEARTBEAT = 'HEARTBEAT';
+export const LOGGER_CATEGORY_FEDERATOR = 'Federators';
```
Comment (pmprete) [inline federator/src/lib/logs.ts:9]: I think we should choose one casing standard? maybe all names should be in caps?

**Review by pmprete — APPROVED** [review]
Awsome

File: federator/src/lib/logs.ts (hunk 2 above)
Comment (RafilxTenfen) [reply federator/src/lib/logs.ts:9]: Agreed

**Pull request merged**